spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryMethod.java

```diff
@@ -504,6 +504,16 @@ public void verify() {
 				}
 			}
 		}
+		if (hasAnnotatedAggregation()) {
+			for (String stage : getAnnotatedAggregation()) {
```

Comment:
How about using `ParameterBindingJsonReader` to verify that each declaration contains a single object?

Reply:
The `QueryMethod` itself does at this point not know about any parameters nor the parser to use so any validation attempt would be quite error prone due to the potential, non resolvable placeholders contained in the source string. 
However there's `BsonUtils#isJsonArray` which we might want to reuse.